medmodels/data_synthesis/mtgan/modules/postprocessor.pyi

```diff
@@ -0,0 +1,59 @@
+from typing import Any, Dict, List, Literal, TypedDict
+
+import sparse
+import torch
+from torch import nn
+from typing_extensions import TypeAlias
+
+from medmodels.data_synthesis.mtgan.modules.preprocessor import PreprocessingAttributes
+from medmodels.medrecord.medrecord import MedRecord
+from medmodels.medrecord.types import MedRecordAttribute, NodeIndex
+
+AttributeType: TypeAlias = Literal["categorical", "continuous", "temporal"]
+
+class PostprocessingHyperparameters(TypedDict, total=False):
+    number_patients_generated: int
+    training_epochs: int
+    hidden_dim: int
+    learning_rate: float
+    number_layers: int
+    batch_size: int
+    number_previous_admissions: int
+    top_k_codes: int
+
+class PostprocessingHyperparametersTotal(TypedDict, total=True):
+    number_patients_generated: int
+    training_epochs: int
+    hidden_dim: int
+    learning_rate: float
+    number_layers: int
+    batch_size: int
+    number_previous_admissions: int
+    top_k_codes: int
```

Comment:
there might be a way to do this without duplicating all keys. I asked chatgpt but can't check if it works right now

```python
# Define the TypedDict with all fields specified
class PostprocessingHyperparameters(TypedDict, total=True):
    number_patients_generated: int
    training_epochs: int
    hidden_dim: int
    learning_rate: float
    number_layers: int
    batch_size: int
    number_previous_admissions: int
    top_k_codes: int

# Optional version of the TypedDict where none of the fields are required
PostprocessingHyperparametersOptional = TypedDict(
    "PostprocessingHyperparametersOptional",
    PostprocessingHyperparameters.__annotations__,
    total=False,
)```

This is also not super nice but at least we wont have to keep track of all the hyperparameters in two places

Reply:
This does not work really:
_Expected dict or keyword parameter as second parameter._

Tried a couple more things and it is not really working:/ We tried but it looks like there is no way without repetition. I will implement a test to check that all the fields have the same name at least.